src/sha256.nr

```diff
@@ -512,6 +513,278 @@ fn hash_final_block(msg_block: MSG_BLOCK, mut state: STATE) -> HASH {
     out_h
 }
 
+/**
+ * Partially computes a SHA256 hash of a message but does not finalize
+ * @notice can be used for post-partial hashing where client proves part of hash and relies on server to finish
+ * 
+ *
+ * @param N: the length of the message to hash.
+ *  --- WARNING: N must be divisible by BLOCK_SIZE such that N % BLOCK_SIZE == 0 
+ *      otherwise the remaining bytes will not be inputted when computing the initial hash
+ * @param msg: the preimage to begin hashing
+ * @return the intermediate hash state
+ */
+pub fn partial_sha256_var_start<let N: u32>(msg: [u8; N], message_size: u32) -> [u32; 8] {
```

Comment:
why not use `BoundedVec` here & other fns?
```rs
pub fn partial_sha256_var_start<let N: u32>(msg: BoundedVec<u8, N>) -> [u32; 8] {
```

Reply:
This can be done in a followup